cloud: add SQL and TiFlash related docs (#9001) (#9479)
pingcap/docs: system-variables.md

### ddl_slow_threshold

<CustomContent platform="tidb-cloud">

> **Note:**
>

> > This TiDB variable is not applicable to TiDB Cloud.

Reviewer: Ditto.

### tidb_allow_remove_auto_inc <span class="version-mark">New in v2.1.18 and v3.0.4</span>

<CustomContent platform="tidb-cloud">

> **Note:**
>

> > This TiDB variable is not applicable to TiDB Cloud.

Reviewer: Ditto.

</CustomContent>

<CustomContent platform="tidb-cloud">

- This variable specifies the way of scanning locks in the Resolve Locks step of GC. When the variable value is set to `LEGACY`, TiDB scans locks by Regions. When the value `PHYSICAL` is used, it enables each TiKV node to bypass the Raft layer and directly scan data, which can effectively mitigate the impact of GC wakening up all Regions, thus improving the execution speed in the Resolve Locks step.

</CustomContent>

### tidb_general_log

<CustomContent platform="tidb-cloud">

> **Note:**
>
> This TiDB variable is not applicable to TiDB Cloud.

</CustomContent>

- Scope: GLOBAL
- Persists to cluster: No
- Type: Boolean
- Default value: `OFF`

<CustomContent platform="tidb-cloud">

> - This variable is used to set whether to record all SQL statements in the log. This feature is disabled by default. If maintenance personnel needs to trace all SQL statements when locating issues, they can enable this feature.

Reviewer: Looks like we can also use the second person here?
```suggestion
- This variable is used to set whether to record all SQL statements in the log. This feature is disabled by default. If you need to trace all SQL statements when locating issues, enable this feature.
```

- Type: Integer
- Default value: `0`
- Range: `[0, 2147483647]`

<CustomContent platform="tidb">

- This variable is used to adjust the maximum days of logger on the current TiDB instance. Its value defaults to the value of the [`max-days`](/tidb-configuration-file.md#max-days) configuration in the configuration file. Changing the variable value only affects the current TiDB instance. After TiDB is restarted, the variable value is reset and the configuration value is not affected.

</CustomContent>

<CustomContent platform="tidb-cloud">

> - This variable is used to adjust the maximum days of logger on the current TiDB instance. Changing the variable value only affects the current TiDB instance. After TiDB is restarted, the variable value is reset and the configuration value is not affected.

Reviewer: A little bit confused. Please confirm whether I understand it right.
```suggestion
- This variable is used to adjust the maximum days of logger on the current TiDB instance. Changing the variable value only affects the current TiDB instance. After TiDB is restarted, the variable value is reset and the original configuration value is not affected.
```

Reviewer 2: The configuration value here means "the value of the [`max-days`](/tidb-configuration-file.md#max-days) configuration in the configuration file". To avoid confusion, let me remove those explanationson for TiDB Cloud because this variable is not applicable to TiDB Cloud.

> - This variable is used to adjust the maximum days of logger on the current TiDB instance. Changing the variable value only affects the current TiDB instance. After TiDB is restarted, the variable value is reset and the configuration value is not affected.

Reviewer 2: ```suggestion
- This variable is used to adjust the maximum days of logger on the current TiDB instance.
```

- Unit: Bytes
- This variable is used to set the threshold value of memory quota for a query.
- If the memory quota of a query during execution exceeds the threshold value, TiDB performs the operation designated by `tidb_mem_oom_action`.

<CustomContent platform="tidb">

> - This setting was previously session scoped and used the value of `mem-quota-query` from `tidb.toml` as an initial value. Starting from v6.1.0, `tidb_mem_quota_query` is now a `SESSION | GLOBAL` scoped variable.

Reviewer: ```suggestion
- For versions earlier than TiDB v6.1.0, this is a session scope variable and uses the value of `mem-quota-query` from `tidb.toml` as an initial value. Starting from v6.1.0, `tidb_mem_quota_query` is a `SESSION | GLOBAL` scope variable.
```

</CustomContent>

<CustomContent platform="tidb-cloud">

> - This setting was previously session scoped. Starting from v6.1.0, `tidb_mem_quota_query` is now a `SESSION | GLOBAL` scoped variable.

Reviewer: ```suggestion
- For versions earlier than TiDB v6.1.0, this is a session scope variable. Starting from v6.1.0, `tidb_mem_quota_query` is a `SESSION | GLOBAL` scope variable.
```
